packages/react-reconciler/src/ReactFiberThrow.js

```diff
@@ -0,0 +1,383 @@
+          let retry = resolveRetryThenable.bind(null, workInProgress, thenable);
+          if (enableSchedulerTracing) {
+            retry = Schedule_tracing_wrap(retry);
+          }
+          thenable.then(retry, retry);
+        }
+        workInProgress.effectTag |= ShouldCapture;
+        workInProgress.expirationTime = renderExpirationTime;
+        return;
+      }
+      // This boundary already captured during this render. Continue to the next
+      // boundary.
+      workInProgress = workInProgress.return;
+    } while (workInProgress !== null);
+    // No boundary was found. Fallthrough to error mode.
+    // TODO: Use invariant so the message is stripped in prod?
+    value = new Error(
+      (getComponentName(sourceFiber.type) || 'A React component') +
+        ' suspended while rendering, but no fallback UI was specified.\n' +
+        '\n' +
+        'Add a <Suspense fallback=...> component higher in the tree to ' +
+        'provide a loading indicator or placeholder to display.' +
+        getStackByFiberInDevAndProd(sourceFiber),
+    );
+  }
+
+  // We didn't find a boundary that could handle this type of exception. Start
+  // over and traverse parent path again, this time treating the exception
+  // as an error.
+  renderDidError();
+  value = createCapturedValue(value, sourceFiber);
+  let workInProgress = returnFiber;
+  do {
+    switch (workInProgress.tag) {
+      case HostRoot: {
+        const errorInfo = value;
+        workInProgress.effectTag |= ShouldCapture;
+        workInProgress.expirationTime = renderExpirationTime;
+        const update = createRootErrorUpdate(
+          workInProgress,
+          errorInfo,
+          renderExpirationTime,
+        );
+        enqueueCapturedUpdate(workInProgress, update);
+        return;
+      }
+      case ClassComponent:
+        // Capture and retry
+        const errorInfo = value;
+        const ctor = workInProgress.type;
+        const instance = workInProgress.stateNode;
+        if (
+          (workInProgress.effectTag & DidCapture) === NoEffect &&
+          (typeof ctor.getDerivedStateFromError === 'function' ||
+            (instance !== null &&
+              typeof instance.componentDidCatch === 'function' &&
+              !isAlreadyFailedLegacyErrorBoundary(instance)))
+        ) {
+          workInProgress.effectTag |= ShouldCapture;
+          workInProgress.expirationTime = renderExpirationTime;
+          // Schedule the error boundary to re-render using updated state
+          const update = createClassErrorUpdate(
+            workInProgress,
+            errorInfo,
+            renderExpirationTime,
+          );
+          enqueueCapturedUpdate(workInProgress, update);
+          return;
+        }
+        break;
+      default:
+        break;
+    }
+    workInProgress = workInProgress.return;
+  } while (workInProgress !== null);
+}
+
+export {throwException, createRootErrorUpdate, createClassErrorUpdate};
```

Comment:
Any reason for putting the exports at the bottom? We're really inconsistent about this. Maybe we should pick one and stick with it.

Reply:
This is just copy pasta so at least it's consistent with what it was.

I think the primary reason exports at the bottom are left over from the CommonJS form.